### PR TITLE
feat: create public api

### PR DESCRIPTION
**Pull request by thatfiredev**
This PR should create our public API and add the dependency on retrofit for HTTP requests.

### CODE REVIEW

**Comment by edsonmichaque** [comment]
@rosariopfernandes do we expect to allow the modification of configurations after creating an instance of Client?
I am wondering if it's useful to expose them to end user

**Comment by thatfiredev** [comment]
@edsonmichaque No, a Client is not supposed to be modified. What are you suggesting?

**Comment by edsonmichaque** [comment]
@rosariopfernandes, I think the Client interface doesn't need getters and setter for configuration properties. We can have only receive, send, query and revert methods.

If you take a look at the other SDK's you will notice a  **Configuration** class, maybe we can extract configuration properties to another class or just omit getters and setters

**Comment by thatfiredev** [comment]
@edsonmichaque the getters are for our own use internally and the `Client` class has no setters. The setters are present in the `Builder`.

**Comment by edsonmichaque** [comment]
@rosariopfernandes, my bad, about the setters
But Client is the user facing class, so anyone call call the getters

**Comment by thatfiredev** [comment]
@edsonmichaque Yes. Thought the getters would be useful (in the future) for developers who may want to extend our SDK in any kind of way.

But maybe we can drop them for now and add them later, if necessary.

**Comment by edsonmichaque** [comment]
@rosariopfernandes I agree, when we need them we will add them

**Comment by thatfiredev** [comment]
@edsonmichaque I've updated the PR accordingly. PTAL

**Comment by edsonmichaque** [comment]
@rosariopfernandes, maybe we dont need to declare an Environment member in client, but we can still  receive it in builder to fill host member variable

 Something like this:

```java
if (environment == Environment.DEVELOPMENT) {
    host = "<sandbox url>"
} else if (environment == Environment.PRODUCTION) {
    host = "<production url>"
} else {
    ///throw an exception
}
```

But the end user may prefer to fill the host directly, if it happens we can ignore the environment member

File: src/main/java/org/paymentsds/mpesa/Client.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    private final Environment environment;
+    private final String host;
+    private final String securityCredential;
+
+    private Client(
+            String apiKey,
+            String publicKey,
+            String serviceProviderCode,
+            String initiatorIdentifier,
+            Environment environment,
```
Comment (edsonmichaque) [inline src/main/java/org/paymentsds/mpesa/Client.java]: I think we can remove this property, we just need it to fill the host name

File: src/main/java/org/paymentsds/mpesa/Client.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            }
+            if (publicKey == null) {
+                throw new IllegalArgumentException("Client must contain a publicKey");
+            }
+            if (host == null) {
+                throw new IllegalArgumentException("Client must contain a host");
+            }
+            if (environment == null) {
+                environment = Environment.DEVELOPMENT;
+            }
```
Comment (edsonmichaque) [inline src/main/java/org/paymentsds/mpesa/Client.java]: We whether need a host or an environment, I think we shouldn't validate them separately

**Review by edsonmichaque — COMMENTED** [review]
@rosariopfernandes, I am happy, we have just to improve host vs environment validation, apart from this we can merge

File: src/main/java/org/paymentsds/mpesa/Client.java (hunk 1 above)
Comment (thatfiredev) [reply src/main/java/org/paymentsds/mpesa/Client.java]: Sounds good. I'll remove the Environment property

File: src/main/java/org/paymentsds/mpesa/Client.java (hunk 2 above)
Comment (thatfiredev) [reply src/main/java/org/paymentsds/mpesa/Client.java]: I've never used the API in production. What's the host url in that case?

I can see the sandbox is `api.sandbox.vm.co.mz`

Comment (edsonmichaque) [reply src/main/java/org/paymentsds/mpesa/Client.java]: I think it's api.vm.co.mz

Comment (thatfiredev) [reply src/main/java/org/paymentsds/mpesa/Client.java]: @edsonmichaque Thanks!

**Pull request merged**